aiida/cmdline/utils/pluginable.py

```diff
@@ -22,20 +22,36 @@ class Pluginable(click.Group):
 
     def __init__(self, *args, **kwargs):
         """Initialize with entry point group."""
+        self._isolated_mode = False
         self._entry_point_group = kwargs.pop('entry_point_group')
         super(Pluginable, self).__init__(*args, **kwargs)
 
     def list_commands(self, ctx):
         """Add entry point names of available plugins to the command list."""
         subcommands = super(Pluginable, self).list_commands(ctx)
-        subcommands.extend(get_entry_point_names(self._entry_point_group))
+
+        if not self._isolated_mode:
+            subcommands.extend(get_entry_point_names(self._entry_point_group))
+
         return subcommands
 
     def get_command(self, ctx, name):  # pylint: disable=arguments-differ
         """Try to load a subcommand from entry points, else defer to super."""
         command = None
-        try:
-            command = load_entry_point(self._entry_point_group, name)
-        except exceptions.EntryPointError:
+        if not self._isolated_mode:
+            try:
+                command = load_entry_point(self._entry_point_group, name)
+            except exceptions.EntryPointError:
+                command = super(Pluginable, self).get_command(ctx, name)
+        else:
             command = super(Pluginable, self).get_command(ctx, name)
         return command
+
+    def set_isolated_mode(self, isolated_mode):
```

Comment:
what about calling the variable something like `include_external_plugins` instead of `isolated_mode`?

Reply:
`exclude_external_plugins` would make more sense, no? Since the normal behavior should be to include it. The excluding of them is the exceptional behavior.